src/components/message-display-rules/DisplayRule.tsx

```diff
@@ -0,0 +1,142 @@
+/** *****************************************************************************
+ * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************** */
+
+import React, { useRef, useState } from 'react';
+import { createStyleSelector } from '../../helpers/styleCreators';
+import { useMessageDisplayRulesStore, useOutsideClickListener } from '../../hooks';
+import { MessageDisplayRule } from '../../models/EventMessage';
+import NewRuleForm from './NewRuleForm';
+
+interface DisplayRuleProps {
+	rule: MessageDisplayRule | null;
+	sessions: string[];
+	index: number;
+	isFirst: boolean | null;
+	isLast: boolean | null;
+}
+
+interface DisplayRuleRendererProps {
+	rule: MessageDisplayRule | null;
+	isFirst: boolean | null;
+	isLast: boolean | null;
+	index: number;
+}
+
+const DisplayRuleRenderer = ({ rule, isFirst, isLast, index }: DisplayRuleRendererProps) => {
+	const rulesStore = useMessageDisplayRulesStore();
+	if (!rule) {
+		return null;
+	}
+	const renderReorder = () => {
+		if (isFirst === null && isLast === null) return null;
+		if (isFirst && isLast) {
+			return null;
+		}
+		if (isFirst) {
+			return (
+				<div className='reorder'>
+					<button
+						className='reorder-control down'
+						onClick={() => {
+							rulesStore.reorderMessagesDisplayRule(index, index + 1);
+						}}
+					/>
+				</div>
+			);
+		}
+		if (isLast) {
+			return (
+				<div className='reorder'>
+					<button
+						className='reorder-control up'
+						onClick={() => {
+							rulesStore.reorderMessagesDisplayRule(index, index - 1);
+						}}
+					/>
+				</div>
+			);
+		}
+		return (
+			<div className='reorder'>
```

Comment:
why not return 
`
<div>
 {!isFirst && <button>Up<button>}
 {is!Last && <button>Down</button>}
</div>
`